nrf_wifi/fw_if/umac_if/src/default/fmac_api.c

```diff
@@ -440,6 +440,44 @@ void nrf_wifi_fmac_dev_deinit(struct nrf_wifi_fmac_dev_ctx *fmac_dev_ctx)
 	nrf_wifi_hal_unlock_rx(fmac_dev_ctx->hal_dev_ctx);
 }
 
+#ifdef CONFIG_NRF_WIFI_RPU_RECOVERY
+enum nrf_wifi_status nrf_wifi_fmac_rpu_recovery_callback(void *mac_dev_ctx,
+						void *event_data,
+						unsigned int len)
+{
+	enum nrf_wifi_status status = NRF_WIFI_STATUS_FAIL;
+	struct nrf_wifi_fmac_dev_ctx *fmac_dev_ctx = NULL;
+	struct nrf_wifi_fmac_dev_ctx_def *def_dev_ctx = NULL;
+	struct nrf_wifi_fmac_priv *fpriv = NULL;
+	struct nrf_wifi_fmac_priv_def *def_priv = NULL;
+
+	fmac_dev_ctx = mac_dev_ctx;
+	if (!fmac_dev_ctx) {
+		nrf_wifi_osal_log_err(fpriv->opriv,
```

Comment:
`fpriv` has not been initialized yet.

Reply:
Even if it's initialized this is a NULL check for `fmac_dev_ctx` so it won't work, I had already raised a bug to fix all such instances: https://nordicsemi.atlassian.net/browse/SHEL-2639, I will move the assignment up but it will crash, we can take care of that later.